src/appleseed.studio/utility/miscellaneous.cpp

```diff
@@ -187,13 +187,18 @@ QString get_extension(ParamArray& settings, const QString& target_dialog)
 {
     QString filter = get_value(settings, target_dialog + SETTINGS_SELECTED_FILTER);
 
-    const int ext_start = filter.lastIndexOf('(') + 2;
+    const int ext_start = filter.lastIndexOf('(') + 1;
     const int ext_end = filter.lastIndexOf(')');
 
-    if (ext_start != string::npos && ext_end != string::npos && ext_start < ext_end)
-        return filter.mid(ext_start, ext_end - ext_start);
+    if (ext_start == -1 || ext_end == -1 || ext_start > ext_end)
+        return "";
+
+    QStringList extensions = filter.mid(ext_start, ext_end - ext_start).split(' ');
+
+    if (extensions.length() == 1 && extensions[0] != "*.*")
+        return extensions[0].mid(1);
     else
-        return QString("");
+        return ".appleseed";
```

Comment:
I think we might have discussed this some days or weeks ago, but I'm actually a bit surprised that we need to do this at all.

Do you confirm that the file dialog does not append a file extension automatically?

What happens in other situations (such as saving a render as an image file on disk) where we don't do anything special, is an extension appended if it's missing?